File: frontend/lib/supabaseClient.js
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

// Keep static builds usable without exposing or inventing credentials. Auth
// actions surface a clear configuration error when the client is unavailable.
export const supabase =
  supabaseUrl && supabaseAnonKey
    ? createClient(supabaseUrl, supabaseAnonKey)
    : null;

File: backend/app/supabase_client.py
from supabase import Client, create_client

from app.config import settings


supabase: Client | None = (
    create_client(settings.supabase_url, settings.supabase_service_role_key)
    if settings.supabase_url and settings.supabase_service_role_key
    else None
)
